poetry.lock

```diff
@@ -356,7 +356,7 @@ python-versions = ">=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*, !=3.5.*"
 certifi = ">=2017.4.17"
 charset-normalizer = {version = ">=2.0.0,<2.1.0", markers = "python_version >= \"3\""}
 idna = {version = ">=2.5,<4", markers = "python_version >= \"3\""}
-urllib3 = ">=1.21.1,<1.27"
+urllib3 = ">=1.26.5,<1.27"
```

Comment:
Looks like this file was modified manually and broke the solved dependencies. The way to do it is but updating the dependencies with poetry, ie: `poetry update urllib3`. Poetry will update the lock always with the latest possible version given the constraints, BUT I see urllib3 is used requests which is used by sphinx.

Since we have many outdated dependencies I would rather do a `poetry update`, should have an output like:

```
  • Removing toml (0.10.2)
  • Updating certifi (2021.10.8 -> 2022.6.15)
  • Updating charset-normalizer (2.0.9 -> 2.0.12)
  • Updating markupsafe (2.0.1 -> 2.1.1)
  • Updating pyparsing (3.0.6 -> 3.0.9)
  • Updating pytz (2021.3 -> 2022.1)
  • Updating babel (2.9.1 -> 2.10.3)
  • Updating imagesize (1.3.0 -> 1.4.1)
  • Updating jinja2 (3.0.3 -> 3.1.2)
  • Updating pygments (2.10.0 -> 2.12.0)
  • Updating requests (2.26.0 -> 2.27.1)
  • Updating soupsieve (2.3.1 -> 2.3.2.post1)
  • Updating tornado (6.1 -> 6.2)
  • Updating beautifulsoup4 (4.10.0 -> 4.11.1)
  • Updating colorama (0.4.4 -> 0.4.5)
  • Updating restructuredtext-lint (1.3.2 -> 1.4.0)
  • Updating ruamel.yaml (0.17.19 -> 0.17.21)
  • Installing sphinx-basic-ng (0.0.1a12)
  • Installing tomli (2.0.1)
  • Updating blessed (1.19.0 -> 1.19.1)
  • Updating flake8-rst-docstrings (0.2.5 -> 0.2.7)
  • Updating furo (2021.11.23 -> 2022.6.21)
  • Updating pre-commit-hooks (4.1.0 -> 4.3.0)
  • Updating readchar (3.0.4 -> 3.0.5)
```

You can use the info to upgrade the minimum requirements on pyproject.toml and then run `poetry lock`. Ie.:

```[tool.poetry.dependencies]
python = ">=3.7"
blessed = ">=1.19.1"
python-editor = ">=1.0.4"
readchar = ">=3.0.5"

[tool.poetry.dev-dependencies]
pexpect = ">=4.8.0"
Sphinx = ">=4.3.2"
sphinx-autobuild = ">=2021.3.14"
furo = ">=2022.6.21"
flake8 = ">=4.0.1"
flake8-docstrings = ">=1.6.0"
flake8-rst-docstrings = ">=0.2.7"
pre-commit-hooks = ">=4.3.0"
```
 
In the end it would update the urllib3 version to 2.27.1. So then you run CI/CD to check if all is good (which should).

Reply:
To avoid doing this updates manually we can use dependabot to automate the work and create PRs for us.

Reply:
thanks, @staticdev . I'm not used to poetry.
I tried to do a `poetry update`, but updated too many packages. `poetry update urllib3` updatd too many packages as well, and that's why I tried to update it by hand.

Instead, I created a PR #172 to enable dependabot to create these PRs automatically.